state/syncer/userAccountsSyncer.go

```diff
@@ -229,13 +174,65 @@ func (u *userAccountsSyncer) findAllAccountRootHashes(mainTrie common.Trie) ([][
 			continue
 		}
 
-		if len(account.RootHash) > 0 {
-			rootHashes = append(rootHashes, account.RootHash)
-			atomic.AddInt32(&u.numMaxTries, 1)
+		if len(account.RootHash) == 0 {
+			continue
+		}
+
+		err = u.checkGoRoutinesThrottler(ctx)
+		if err != nil {
+			return err
+		}
+
+		//throttler.StartProcessing is required to be here as it prevent the following edge-case:
+		//loop does 100k iterations because throttler.CanProcess allows it and starts 100k go routines that can
+		//not execute their first instructions that could tell the throttler they have started.
+		//Telling the throttler the processing has started here will prevent OOM exception because of too many go
+		//routines started.
+		u.throttler.StartProcessing()
+
+		wg.Add(1)
+
+		go func(trieRootHash []byte) {
+			defer u.throttler.EndProcessing()
+
+			log.Trace("sync data trie", "roothash", trieRootHash)
+			newErr := u.syncDataTrie(trieRootHash, ssh, ctx)
+			if newErr != nil {
+				errMutex.Lock()
+				errFound = newErr
+				errMutex.Unlock()
+			}
+			atomic.AddInt32(&u.numTriesSynced, 1)
+			log.Trace("finished sync data trie", "roothash", trieRootHash)
+			wg.Done()
+		}(account.RootHash)
+
+		atomic.AddInt32(&u.numMaxTries, 1)
```

Comment:
please move this on L194 as it makes more sense

Reply:
Done.